src/vs/workbench/common/editor/editorGroup.ts

```diff
@@ -663,6 +664,14 @@ export class EditorGroup extends Disposable {
 					&& !this.configurationService.getValue<boolean>('sql.promptToSaveGeneratedFiles')) {
 					return;
 				}
+				// Do not add generated files from Temp if file is not dirty
+				if (e instanceof FileEditorInput && !e.isDirty()) {
+					let filePath = e.getResource() ? e.getResource().fsPath : undefined;
+					let tempPath = process.env[process.platform === 'win32' ? 'TEMP' : 'TMPDIR'];
+					if (filePath && tempPath && filePath.toLocaleLowerCase().includes(tempPath.toLocaleLowerCase())) {
```

Comment:
Any particular reason you aren't you using os.tmpdir()?

Reply:
I tried using - didn't work

Reply:
What do you mean "didn't work"? What didn't work about it?

Reply:
came as undefined

Reply:
Can you please let me know why using process.env is not preferred?

Reply:
Because os.tmpdir does more thourough checking to try and get a valid temp dir since the exact env var may not necessarily be the single one you're checking :


```
if (isWindows) {
		path = process.env.TEMP ||
			process.env.TMP ||
			(process.env.SystemRoot || process.env.windir) + '\\temp';
	} else {
		path = process.env.TMPDIR ||
			process.env.TMP ||
			process.env.TEMP ||
			'/tmp';
	}

	if (trailingSlashRe.test(path)) {
		path = path.slice(0, -1);
	}
```

Reply:
Let me give it another try

Reply:
Looks like it needed me to git clean and restart my machine huh. Thanks for the inspiration :)